src/components/common/Tag.tsx

```diff
@@ -0,0 +1,27 @@
+import type { PropsWithChildren } from 'react';
+
+export interface TagProps {
+  tagStyle?: string;
+}
+
+const Tag = ({ tagStyle = 'dark', children }: PropsWithChildren<TagProps>) => {
+  const darkTagClass = 'text-white bg-gray-500';
+  const mediumTagClass = 'text-gray-500 bg-gray-200';
+  const lightTagClass = 'text-gray-500 bg-white border border-gray-200';
+
+  const defaultTagClass = 'text-b4 px-10 py-4 rounded-full';
+
+  const getStyleClass = () => {
+    if (tagStyle === 'dark') return `${darkTagClass} ${defaultTagClass}`;
+    if (tagStyle === 'medium') return `${mediumTagClass} ${defaultTagClass}`;
+    if (tagStyle === 'light') return `${lightTagClass} ${defaultTagClass}`;
+  };
+
+  return (
+    <div className="flex">
+      <div className={`${getStyleClass()}`}>{children}</div>
+    </div>
+  );
+};
```

Comment:
```suggestion
interface StyleProps {
  [color: string]: string;
}

const styleClass: StyleProps = {
  dark: 'text-white bg-gray-500',
  medium: 'text-gray-500 bg-gray-200',
  light: 'text-gray-500 bg-white border border-gray-200',
  default: 'text-b4 px-10 py-4 rounded-full',
};

const Tag = ({ tagStyle = 'dark', children }: PropsWithChildren<TagProps>) => {
  return (
    <div className="flex">
      <div className={`${styleClass[tagStyle]} ${styleClass['default']}`}>{children}</div>
    </div>
  );
};
```

style class를 react component 외부에서 관리하고 styleClass를 얻기위한 함수 및 반복문을 제거하기 위해서 obj을 사용해봤습니다!

Reply:
typeof 사용하시면 추론 가능할거예요!

Reply:
type enum 사용해서 상수로 관리하면 될것 같습니다! 

이전에 사용할때 이렇게 많이 적용했어서, 아래와 같은식으로 적용해보도록하겠습니다! 

```
const TagStyleType = {
  LIGHT: 'LIGHT',
  MEDIUM: 'MEDIUM',
  DARK: 'DARK',
} as const;
```

Reply:
타입 상수 사용하여 수정했습니다!
https://github.com/depromeet/pingpong-client/pull/42/commits/79aaa3e3d3a488166d1b8aeb727e3f53aa6830c4
https://github.com/depromeet/pingpong-client/pull/42/commits/66b74aef36f51da0936f26ea2cb6f10fbd2f0f1e